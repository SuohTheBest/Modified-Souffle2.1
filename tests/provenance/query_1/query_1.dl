// Souffle - A Datalog Compiler
// Copyright (c) 2017, The Souffle Developers. All rights reserved
// Licensed under the Universal Permissive License v 1.0 as shown at:
// - https://opensource.org/licenses/UPL
// - <souffle root>/licenses/SOUFFLE-UPL.txt

// This code tests the provenance explain interface for a simple path example.

.pragma "provenance" "explain"

.decl edge(x:symbol, y:symbol)
edge("start", "B0").
edge("B0", "B1").
edge("B1", "B2").
edge("B2", "B4").
edge("B2", "B6").
edge("B3", "B4").
edge("B4", "B4").
edge("B4", "B5").
edge("B5", "B6").
edge("B6", "B1").
edge("B6", "end").

.decl path(x:symbol, y:symbol)
path(x, y) :- edge(x, y).
path(x, z) :- edge(x, y), path(y, z).
.output path()
