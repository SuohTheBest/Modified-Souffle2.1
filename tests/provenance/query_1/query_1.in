query path("start", x), path(x, "end")
;
;
;
;
;
exit
