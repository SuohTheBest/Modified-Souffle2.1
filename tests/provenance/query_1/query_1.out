x = B0
x = B1
x = B2
x = B4
x = B6
x = B5.
